influx-reports/influx.py

```diff
@@ -0,0 +1,166 @@
+
+    client = InfluxDBClient(influx_ip, influx_port, database='telegraf')
+
+    # First find the first and last timestamp from telegraf.
+    # This way querying maas_* data will always
+    # be accurate (if no data is reported by maas plugins,
+    # this should be a failure metric)
+    find_first_timestamp_query = ("select first(total) from processes "
+                                  "where job_reference = '{}';".format(job_reference))
+    first_ts = return_time(client, find_first_timestamp_query, delta_seconds=5)
+    find_last_timestamp_query = ("select last(total) from processes "
+                                 "where job_reference = '{}';".format(job_reference))
+    last_ts = return_time(client, find_last_timestamp_query, delta_seconds=-5)
+
+    logging.info(
+        "Metrics were gathered between {} and {}".format(first_ts, last_ts))
+
+    measurements = ['maas_glance', 'maas_cinder', 'maas_keystone', 'maas_heat', 'maas_neutron',
+                    'maas_nova', 'maas_horizon']
+
+    yml_report = False
+    subunit_report = False
+
+    if args.ymlreport:
+        yml_report = True
+        output_file = open(args.ymlreport, 'w+')
+        output_file.write("---\n")
+    if args.subunitreport:
+        subunit_report = True
+        output_stream = open(args.subunitreport, 'wb')
+        output = StreamResultToBytes(output_stream)
+        output.startTestRun()
+
+    for measurement in measurements:
+        measurement_data = return_uptime(client, measurement, job_reference)
+        yml_output = yaml.safe_dump(
+            {measurement: dict(measurement_data)}, default_flow_style=False)
+        logging.info(yml_output)
+        if yml_report:
+            output_file.write(yml_output)
+        if subunit_report:
+            for measure in measurement_data:
+                # Input status
+                output.status(
+                    test_id=measure,
+                    timestamp=first_ts
+                )
+
+                # Output the end of the event
+                output.status(
+                    test_id=measure,
+                    # TODO(evrardjp): Define a Threshold to compare with
+                    # and use it as a success criteria.
+                    test_status="success",
+                    test_tags=None,
+                    runnable=False,
+                    # file_name=measure,
+                    # file_bytes=str(measurement_data[measure]).encode('utf-8'),
+                    timestamp=last_ts,
+                    eof=True,
+                    mime_type='text/plain; charset=UTF8'
+                )
+    if subunit_report:
+        output.stopTestRun()
+        output_stream.close()
+    if yml_report:
+        output_file.close()
+
+
+if __name__ == "__main__":
+    """ Args parser and logic router """
+    logging.getLogger().setLevel(logging.INFO)
+    parser = argparse.ArgumentParser(
+        description='Fetch maas_ metrics, and report downtime data.')
```

Comment:
The file docstring refers to uptime, this description refers to downtime. More generally, the script help text and the YAML output file provide little/no information regarding what the output actually is.

The following is part of a YAML report file I've generated:
```
---
maas_glance:
  glance_api_local_status: 1740
  glance_registry_local_status: 1800
maas_cinder:
  cinder-backup_status: 180
  cinder-scheduler_status: 240
  cinder-volume-lvm_status: 180
  cinder_api_local_status: 1740
```

What do the numbers mean? What are the units? I think it would be useful to include a comment at the top of the YAML file explaining what the data is.